packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -131,7 +87,14 @@ export class DbMutationQueue {
      * After sending this token, earlier tokens may not be used anymore so
      * only a single stream token is retained.
      */
-    public lastStreamToken: string
+    public lastStreamToken: string,
+    /**
+     * An identifier for the highest numbered batch that has been acknowledged
+     * by the server. All MutationBatches in this queue with batchIds less
+     * than or equal to this value are considered to have been acknowledged by
+     * the server.
+     */
+    public highestPendingBatchId: number
```

Comment:
This only exists so that we can leave acknowledged mutations in the queue for the sake of other tabs being able to read them, right?  I think it's worth calling that out here. Also, are we going to port this to other platforms?  If not, please add a "// PORTING NOTE:" comment.

Reply:
I updated the outdated comment. This is to reduce the number of scans we have to do in the mutation batch. Right now, to get the highest batch ID, we do an full scan at startup. I don't want to do this every time we add a mutation from a secondary tab.